site/src/hooks/useClassName.ts

```diff
@@ -13,9 +15,14 @@ export type ClassName = (cssFn: typeof css, theme: Theme) => string;
  */
 export function useClassName(styles: ClassName, deps: DependencyList): string {
   const theme = useTheme();
-  const className = useMemo(() => {
-    return styles(css, theme);
-  }, [...deps, theme]);
+  const stableStylesCallback = useEffectEvent(styles);
```

Comment:
what do we gain by having a stable function identity, when the `useMemo` won't be recomputed?

Reply:
...And speaking of fragile code, my change actually makes things worse.

The idea was something I do a lot in TypeScript, which is "limiting the surface area of my lies". As in, when working with a lot of type generics, I'll get the type 95% right, and then assert it as a generic to avoid too much drift. With this, the idea was that I would get the dependency array 95% right, and then disable the rule to handle the things that can't be handled via static analysis

But in this case, because of React order of operations, `useMemo` will run before `useLayoutEffect` (which is what powers `useEffectEvent`). So the hook result will now be stale when it wasn't before

I'll add a test case that fails, and update the code to make sure things are working properly